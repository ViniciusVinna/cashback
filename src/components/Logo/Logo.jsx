import React from 'react';
import PropTypes from 'prop-types';

import { ReactComponent as GroupBrandDark } from 'assets/brands/group-logo-dark-mode.svg';
import { ReactComponent as GroupBrandLight } from 'assets/brands/group-logo-light-mode.svg';
import { ReactComponent as GroupSign } from 'assets/brands/sign.svg';

import { LogoStyled } from './Logo.styles';

const { LogoWrapper } = LogoStyled;

const Logo = ({ type }) => {
  let output;

  if (type === 'sign') {
    output = (<GroupSign preserveAspectRatio="xMidYMid slice" />);
  }
  else if (type === 'group-light') {
    output = (<GroupBrandLight preserveAspectRatio="xMidYMid slice" />);
  }
  else if (type === 'group-dark') {
    output = (<GroupBrandDark preserveAspectRatio="xMidYMid slice" />);
  }

  return (
    <LogoWrapper data-testid="logo">
      {output}
    </LogoWrapper>
  );
};

Logo.propTypes = {
  type: PropTypes.string,
};

Logo.defaultProps = {
  type: 'sign',
};

export default Logo;
