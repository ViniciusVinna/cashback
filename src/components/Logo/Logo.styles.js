import styled from '@emotion/styled';

export const LogoStyled = {
  LogoWrapper: styled('div')`
    display: flex;
    justify-content: center;
    align: center;
    width: 100%;
    
    svg {
      height: 100%;
    }
  `,
};
