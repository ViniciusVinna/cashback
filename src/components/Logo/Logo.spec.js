import React from 'react';
import Logo from './Logo';

describe('Logo', () => {
  it('should render the default sign logo when no props is declared', async () => {
    const { container } = render(<Logo />);

    expect(container).toMatchSnapshot();
  });

  it('should render the group-light logo', async () => {
    const { container } = render(<Logo type="group-light" />);

    expect(container).toMatchSnapshot();
  });

  it('should render the group-dark logo', async () => {
    const { container } = render(<Logo type="group-dark" />);

    expect(container).toMatchSnapshot();
  });
});
