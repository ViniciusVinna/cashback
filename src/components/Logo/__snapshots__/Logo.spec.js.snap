// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo should render the default sign logo when no props is declared 1`] = `
<div>
  <div
    class="css-1mcyu8t"
    data-testid="logo"
  >
    <svg
      preserveAspectRatio="xMidYMid slice"
    >
      sign.svg
    </svg>
  </div>
</div>
`;

exports[`Logo should render the group-dark logo 1`] = `
<div>
  <div
    class="css-1mcyu8t"
    data-testid="logo"
  >
    <svg
      preserveAspectRatio="xMidYMid slice"
    >
      group-logo-dark-mode.svg
    </svg>
  </div>
</div>
`;

exports[`Logo should render the group-light logo 1`] = `
<div>
  <div
    class="css-1mcyu8t"
    data-testid="logo"
  >
    <svg
      preserveAspectRatio="xMidYMid slice"
    >
      group-logo-light-mode.svg
    </svg>
  </div>
</div>
`;
